lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -638,7 +638,7 @@ void AMDGPUPassConfig::addIRPasses() {
 
   if (TM.getOptLevel() > CodeGenOpt::None) {
     // ToDo: Fix it so that it can handle addrspacecast to private
-    //addPass(createAMDGPUPromoteAlloca());
+    addPass(createAMDGPUPromoteAlloca());
```

Comment:
Why was this disabled before?

Reply:
I enabled this pass, and ran the HCC unit tests to ensure no regressions; there were no failures.  So adding this pass seems to be ok.  I'm not sure why it was originally disabled.

Reply:
it was disabled quite some time ago for historical reasons i don't recall now